src/types/index.ts

```diff
@@ -1,15 +1,31 @@
 export type Command = () => void
 export type AsyncCommand = () => Promise<void>
 
-export interface TokenDetails {
-  id: number
-  name?: string
+export interface WithAddressMainnet {
+  addressMainnet?: string
+}
+
+export interface WithId {
+  id: string
+}
+
+export interface WithSymbolAndName {
   symbol?: string
-  decimals?: number
+  name?: string
+}
+
+export interface WithAddress {
   address: string
-  addressMainnet?: string
 }
 
-export interface TokenDetailsConfig extends Omit<TokenDetails, 'address'> {
+export interface WithDecimals {
+  decimals: number
+}
+
+export type TokenErc20 = WithSymbolAndName & WithAddress & WithDecimals
+export type Token = TokenErc20 & WithAddressMainnet & Partial<WithId>
```

Comment:
Should we use `Partial<WithAddressMainnet>` for conformity with `WithId` and just have all sub types required
Or as an alternative have all subtypes not required and then use `Required<WithId>`

Reply:
Good point, I thought about this, and I think `addressMainnet` will always be optional. So it was just convenient. 
what about `WithAddressMainnetOpt`? 

I would love to avoid Partial<WithAddressMainnet> for adding this basic prop that only some types have.

Reply:
`WithAddressMainnetOpt` :+1: